src/main/java/eu/europeana/metis/sandbox/service/workflow/TransformationServiceImpl.java

```diff
@@ -41,12 +51,44 @@ public RecordInfo transform(Record record) {
     return new RecordInfo(Record.from(record, recordTransformed));
   }
 
+  @Override
+  public RecordInfo transformToEdmExternal(Record record) {
+    requireNonNull(record, "Record must not be null");
+    String xsltToEdmExternal = datasetRepository.getOne(Integer.valueOf(
+        record.getDatasetId())).getXsltTransformerEdmExternal();
+
+    return new RecordInfo(Record.from(record, transformToEdmExternal(record.getDatasetId(),
+        record.getDatasetName(),
+        new ByteArrayInputStream(xsltToEdmExternal.getBytes(StandardCharsets.UTF_8)),
+        record.getCountry(), record.getLanguage(), record.getContent())));
+  }
+
+  @Override
+  public byte[] transformToEdmExternal(String datasetId, String datasetName, InputStream xsltToEdmExternal,
+      Country country, Language language, byte[] recordContent) {
+
+    byte[] recordToEdmExternal;
+    try {
+      XsltTransformer transformer = new XsltTransformer(getXmlDatasetName(datasetId, datasetName), xsltToEdmExternal,
+          getXmlDatasetName(datasetId, datasetName), country.xmlValue(), language.name().toLowerCase());
+      recordToEdmExternal = transformer.transformToBytes(recordContent, null);
+    } catch (TransformationException e) {
+      throw new RecordProcessingException(datasetId, e);
+    }
+
+    return recordToEdmExternal;
+  }
+
   private XsltTransformer getTransformer(String datasetName, String edmCountry,
       String edmLanguage) {
     return xsltTransformer.getObject(datasetName, edmCountry, edmLanguage);
   }
 
   private String getXmlDatasetName(Record record) {
```

Comment:
I would rename those a bit. It can be confusing, we are getting a datasetName but we already have it from the record information. So what do we get from those methods?

Reply:
I see your point, I've renamed them, not sure if it's better. Let me know what you think of `getNewXmlName`.
If you come up with another idea I'm all ears

Reply:
getJoinDatasetIdDatasetName

Reply:
Done